### ISSUE TITLE
"preResolveTypes" Fragment not respected in GraphQL conditional fragment

### ISSUE THREAD

**Issue by klemenoslaj** [dotansimha/graphql-code-generator#2617]
**Describe the bug**

When having `preResolveTypes` enabled Fragment used within a GraphQL `... on SomeType` is not generated.

*GraphQL*
```graphql
query SomeQuery {
  object {
    id
    ... on SomeType {
      name
      ...FragmentName
    }
  }
}
```

*TypeScript* (_no fragment in the type_)
```typescript
type SomeQuery {
  object: Maybe<(
    | { id: string }
    | { id: string; name: string }
  )>;
}
```

Without the `preResolveTypes` types is correctly generated
*TypeScript*
```typescript
type SomeQuery {
  object: Maybe<(
    | Pick<GlobalType, 'id'>
    | Pick<AnotherType, 'id' | 'name'> & FragmentName
  )>;
}
```

_Please not I wrote all the queries and generated types by hand_

**To Reproduce**

- Create a query with a fragment usage similar to above
- Enable `preResolveTypes`
- Generate the types

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: "http://localhost:4000/graphql"
documents: "./projects/**/*.graphql.ts"
generates:
  ./generated/graphql.d.ts:
    plugins:
      - "typescript"
      - "typescript-operations"
    config:
      avoidOptionals: true
      dedupeOperationSuffix: true
      immutableTypes: true
      # preResolveTypes: true
      nonOptionalTypename: true
      declarationKind: "interface"
  ./generated/schema.json:
    plugins:
      - "introspection"
  ./generated/introspection-result.ts:
    plugins:
      - fragment-matcher
```

**Expected behavior**

**Environment:**

- OS: darwin x64
- graphql-codegen:
@graphql-codegen/cli@1.7.0
@graphql-codegen/core@1.7.0
@graphql-codegen/fragment-matcher@1.7.0
@graphql-codegen/introspection@1.7.0
@graphql-codegen/plugin-helpers@1.7.0
@graphql-codegen/typescript-operations@1.7.0
@graphql-codegen/typescript@1.7.0
@graphql-codegen/visitor-plugin-common@1.7.0
- NodeJS: 10.16.0

**Comment by dotansimha** [comment]
@klemenoslaj Can you please try the latest alpha version? `1.7.1-alpha-1e9152fb.90`

**Comment by klemenoslaj** [comment]
@dotansimha, just checked it, and yes, it's indeed fixed in this alpha version.
Thanks for the great work :+1:

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
